tests/foreman/virtwho/cli/test_esx_sca.py

```diff
@@ -0,0 +1,549 @@
+        form_data['http-proxy'] = https_proxy_name
+        form_data['no-proxy'] = no_proxy
+        virtwho_config = target_sat.cli.VirtWhoConfig.create(form_data)['general-information']
+        result = target_sat.cli.VirtWhoConfig.info({'id': virtwho_config['id']})
+        assert result['http-proxy']['http-proxy-name'] == https_proxy_name
+        assert result['connection']['ignore-proxy'] == no_proxy
+        command = get_configure_command(virtwho_config['id'], module_sca_manifest_org.name)
+        deploy_configure_by_command(
+            command, form_data['hypervisor-type'], org=module_sca_manifest_org.label
+        )
+        get_configure_file(virtwho_config['id'])
+        assert get_configure_option('https_proxy', ETC_VIRTWHO_CONFIG) == https_proxy_url
+        assert get_configure_option('no_proxy', ETC_VIRTWHO_CONFIG) == no_proxy
+
+    @pytest.mark.tier2
+    def test_positive_rhsm_option(
+        self, module_sca_manifest_org, form_data, virtwho_config, target_sat
+    ):
+        """Verify rhsm options in the configure file"
+
+        :id: 2aad374a-c493-4e3c-91e3-60f21181fd29
+
+        :expectedresults:
+            1. rhsm_hostname, rhsm_prefix are expected
+            2. rhsm_username is not a login account
+
+        :CaseLevel: Integration
+
+        :CaseImportance: Medium
+        """
+        config_file = get_configure_file(virtwho_config['id'])
+        command = get_configure_command(virtwho_config['id'], module_sca_manifest_org.name)
+        deploy_configure_by_command(
+            command, form_data['hypervisor-type'], org=module_sca_manifest_org.label
+        )
+        rhsm_username = get_configure_option('rhsm_username', config_file)
+        assert not User.exists(search=('login', rhsm_username))
+        assert get_configure_option('rhsm_hostname', config_file) == target_sat.hostname
+        assert get_configure_option('rhsm_prefix', config_file) == '/rhsm'
+
+    @pytest.mark.tier2
+    def test_positive_post_hypervisors(self, function_org, target_sat):
+        """Post large json file to /rhsm/hypervisors"
+
+        :id: 6d08e37a-ac72-455e-b173-155f376caff9
+
+        :expectedresults:
+            hypervisor/guest json can be posted and the task is success status
+
+        :CaseLevel: Integration
+
+        :customerscenario: true
+
+        :CaseImportance: Medium
+
+        :BZ: 1637042, 1769680
+        """
+        data = hypervisor_json_create(hypervisors=100, guests=10)
+        url = f"{target_sat.url}/rhsm/hypervisors/{function_org.label}"
+        auth = (settings.server.admin_username, settings.server.admin_password)
+        result = requests.post(url, auth=auth, verify=False, json=data)
+        if result.status_code != 200:
+            if "foreman_tasks_sync_task_timeout" in result.text:
+                task_id = re.findall('waiting for task (.*?) to finish', result.text)[-1]
+                target_sat.wait_for_tasks(search_query=f'id = {task_id}', max_tries=10)
```

Comment:
I think you missed asserting the success result code after wait_for_tasks! Please do add !

Reply:
> I think you missed asserting the success result code after wait_for_tasks! Please do add !

@jyejare Jentedra, 200 is the success code what we need to assert.

Reply:
> > I think you missed asserting the success result code after wait_for_tasks! Please do add !
> 
> @jyejare Jentedra, 200 is the success code what we need to assert.

          ``  else:
                assert result.status_code == 200```